### PR TITLE
adding pcre2/10.35 + cache cmake

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **pcre2/10.35**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
The error is 
```
/home/conan/workspace/cci_PR-2522/2/8_16/.conan/data/pcre2/10.35/_/_/build/e87525cbceccfb221428729b47648040db333393/source_subfolder/src/sljit/sljitConfigInternal.h:219:2: error: #error "-mshstk is needed to compile with -fcf-protection"
  219 | #error "-mshstk is needed to compile with -fcf-protection"
      |  ^~~~~
```

Should I add the missing flag or throw invalid configuration since upstream does not support GCC9 ?

**Comment by uilianries** [comment]
I don't think it's related to GCC9, but probably some feature not enabled in the compiler.

File: recipes/pcre2/all/conanfile.py [hunk 1]
```diff
@@ -41,33 +55,45 @@ def configure(self):
...
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def patch_cmake(self):
+        """Patch CMake file to avoid man and share during install stage"""
+
+        cmake_file = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmake_file, "INSTALL(FILES ${man1} DESTINATION man/man1)", "")
+        tools.replace_in_file(cmake_file, "INSTALL(FILES ${man3} DESTINATION man/man3)", "")
+        tools.replace_in_file(cmake_file, "INSTALL(FILES ${html} DESTINATION share/doc/pcre2/html)", "")
```
Comment (uilianries) [inline recipes/pcre2/all/conanfile.py:86]:
```suggestion
```

Keep the old behavior, it's not a problem removing after installing.

Comment (prince-chrismc) [reply recipes/pcre2/all/conanfile.py:86]: its copy paste from the pcre recipe, Ill fix it

**Review by danimtb — APPROVED** [review]

**Pull request merged**